paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -155,33 +152,32 @@ public class PaparazziPlugin : Plugin<Project> {
         task.nonTransitiveRClassEnabled.set(nonTransitiveRClassEnabled)
         task.targetSdkVersion.set(android.targetSdkVersion())
         task.compileSdkVersion.set(android.compileSdkVersion())
-        task.projectResourceDirs.set(
+
+        val localResourcePaths = localResourceDirs
+          .map { layers -> layers.flatten() }
+          .map { dirs -> dirs.map { projectDirectory.relativize(it.asFile) }.asReversed() }
+
+        task.projectResourceDirs.set(localResourcePaths)
+        task.moduleResourceDirs.set(moduleResourceDirs.relativize(projectDirectory))
+        task.aarExplodedDirs.set(aarExplodedDirs.relativize(gradleHomeDir))
+        task.projectAssetDirs.set(
           run {
-            val resourcesComputer = variant.mergeResourcesProvider
-            val extraGeneratedResDirs =
-              resourcesComputer?.map { it.resourcesComputer.extraGeneratedResFolders }
-                ?: project.provider { project.files() }
-            val generatedResOutputDirs =
-              resourcesComputer?.map { it.resourcesComputer.generatedResOutputDir }
-                ?: project.provider { project.files() }
-            extraGeneratedResDirs
-              .zip(project.provider { localResourceDirs }, FileCollection::plus)
-              .zip(generatedResOutputDirs, FileCollection::plus)
-              .flatMap { it.relativize(projectDirectory) }
+            val localAssetPaths = localAssetDirs
+              .map { layers -> layers.flatten() }
+              .map { dirs -> dirs.map { projectDirectory.relativize(it.asFile) }.asReversed() }
+            val moduleAssetPaths = moduleAssetDirs.relativize(projectDirectory)
+            localAssetPaths.zip(moduleAssetPaths, List<String>::plus)
           }
         )
-        task.moduleResourceDirs.set(moduleResourceDirs.relativize(projectDirectory))
-        task.aarExplodedDirs.set(aarExplodedDirs.relativize(gradleHomeDir))
-        task.projectAssetDirs.set(localAssetDirs.plus(moduleAssetDirs).relativize(projectDirectory))
         task.aarAssetDirs.set(aarAssetDirs.relativize(gradleHomeDir))
         task.paparazziResources.set(buildDirectory.file("intermediates/paparazzi/${variant.name}/resources.json"))
       }
 
       val testVariantSlug = testVariant.name.capitalize(Locale.US)
 
       project.plugins.withType(JavaBasePlugin::class.java) {
-        project.tasks.named("compile${testVariantSlug}JavaWithJavac")
-          .configure { it.dependsOn(writeResourcesTask) }
+        project.tasks.named { it == "compile${testVariantSlug}JavaWithJavac" }
+          .configureEach { it.dependsOn(writeResourcesTask) }
```

Comment:
What's the minimum AGP/Gradle you're aiming for? (and where is it documented?)

This syntax is [Gradle 8.6](https://docs.gradle.org/8.6/release-notes.html#build-authoring-improvements) (>= AGP 8.4)

Reply:
correct, >= AGP 8.4